backend/buildeR.Processor/Services/ProcessorService.cs

```diff
@@ -93,9 +94,19 @@ public async Task BuildProjectAsync(ExecutiveBuildDTO build)
             CloneRepository(build.RepositoryUrl, pathToClonedRepository, build.BranchName);
 
             try
-            {
-                var dockerFileContent = GenerateDockerFileContent(build.BuildSteps, build.RepositoryUrl);
-                await CreateDockerFileAsync(dockerFileContent, pathToClonedRepository);
+            
+                {
+                if (build.BuildSteps.Count() != 0 && build.BuildSteps.FirstOrDefault().BuildStepName == "Dockerfile")
```

Comment:
1. "build.BuildSteps.Count() != 0" => "build.BuildSteps.Any()"
2. FirstOrDefault().BuildStepName will throw exception if it's default